### ISSUE TITLE
Adding a package with the same name as the package will add itself as a workspace

### ISSUE THREAD

**Issue by willgriffiths** [yarnpkg/berry#397]
**Describe the bug**

Adding a package with the same name as the package will add itself as a workspace. 

**To Reproduce**

1. mkdir test && cd test
2. yarn init -y
3. yarn add test

```json
{
  "name": "test",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "test": "workspace:."
  }
}
```

**Environment if relevant (please complete the following information):**

 - OS: OSX 
 - Node version v10.9.0
 - Yarn version 2.0.0-rc.0

**Additional context**

A rare edge case. Thought I would report it anyway. :)

**Comment by arcanis** [comment]
Thanks for the report! I'd be enclined to close the issue as WONTFIX tho as it's quite an edge case and it's easily solvable by adding the `-i` flag or explicitly adding the version (`test@npm:...`) 🤔

**Comment by arcanis** [comment]
Fixed by #421 (workspaces won't add themselves to their own dependencies anymore)
